src/PointwiseFunctions/AnalyticData/GrMhd/MagneticFieldLoop.hpp

```diff
@@ -0,0 +1,218 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Minkowski.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/MakeArray.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd {
+namespace AnalyticData {
+
+/*!
+ * \brief Analytic initial data for an advecting magnetic field loop.
```

Comment:
For my own understanding (maybe @carmaza can answer), the advecting magnetic loop is not an analytic solution to the grmhd equations? Why not, if we expect the loop to advect undisturbed?

Reply:
I'm not aware of any extension to arbitrary time. Would the discontinuous boundary conditions at the loop radius allow for one?

If you're thinking of a "boosted loop", I imagine you'd have to adjust the thermodynamic quantities, as well as the velocity (and the Lorentz factor) appropriately, including a "smooth" transition between regions with and without magnetic field. Plus, you'd need to satisfy the B-field induction equation accordingly.

Reply:
@gsb76 this is not an exact solution, but since the magnetic pressure is much less than the thermal pressure the loop is essentially advected across the grid with a small error (you can plug the values into the GRMHD equations to estimate the size of the error)